scripts/src/chartrepomanager/chartrepomanager.py

```diff
@@ -401,204 +311,6 @@ def create_index_from_report(category, ocp_version_range, report_path):
     return chart_entry
 
 
-def set_package_digest(chart_entry):
-    print("[INFO] set package digests.")
-
-    url = chart_entry["urls"][0]
-    head = requests.head(url, allow_redirects=True)
-    print(f"[DEBUG]: tgz url : {url}")
-    print(f"[DEBUG]: response code from head request: {head.status_code}")
-
-    target_digest = ""
-    if head.status_code == 200:
-        response = requests.get(url, allow_redirects=True)
-        print(f"[DEBUG]: response code get request: {response.status_code}")
-        target_digest = hashlib.sha256(response.content).hexdigest()
-        print(f"[DEBUG]: calculated digest : {target_digest}")
-
-    pkg_digest = ""
-    if "digest" in chart_entry:
-        pkg_digest = chart_entry["digest"]
-        print(f"[DEBUG]: digest in report : {pkg_digest}")
-
-    if target_digest:
-        if not pkg_digest:
-            # Digest was computed but not passed
-            chart_entry["digest"] = target_digest
-        elif pkg_digest != target_digest:
-            # Digest was passed and computed but differ
-            raise Exception(
-                "Found an integrity issue. SHA256 digest passed does not match SHA256 digest computed."
-            )
-    elif not pkg_digest:
-        # Digest was not passed and could not be computed
-        raise Exception(
-            "Was unable to compute SHA256 digest, please ensure chart url points to a chart package."
-        )
-
-
-def update_index_and_push(
```

Comment:
For clarity - this logic is what's being replaced by the new step in the workflow, correct?

Reply:
Yes, the new step can (and should probably) be divided into three smaller logical steps:
* Preparing the `git worktree` -> This was done at the beginning of `chartrepomanager` but only used here. It belongs to the "update index" logic. As these are plain git commands, I moved them to the shell.
* Downlading and updating the index data -> This was part of this function and is now the `update-index` script. This is actually the only part that remained as python code.
* Pushing to the index branch -> This was also part of this function. As they are plain git commands, I moved them to the shell.